### PR TITLE
Try Maven 3.8.4 staging

### PR DESCRIPTION
**Pull request by mickaelistria**

### CODE REVIEW

**Comment by mickaelistria** [comment]
@HannesWell What do you think? Should we merge it immediately and release m2e in 2 days max so it gets in SimRel? Or should we wait after SimRel to merge it?

**Review by HannesWell — CHANGES_REQUESTED** [review]
> @HannesWell What do you think? Should we merge it immediately and release m2e in 2 days max so it gets in SimRel? Or should we wait after SimRel to merge it?

Hard to say. As far as I can tell we have tests to detect the regressions of Maven 3.8.2 (the test was already present but it was overseen when merging the corresponding PR, see #326) and of Maven 3.8.3 (added with #350). So we should not be affect by those regressions. However there could be new ones.

To be save IMHO we should not merge this with only little time to test.
Nevertheless I'm in favour to perform a release that brings the current m2e master to SimRel 2021-12, so at least the fixes/improvements mentioned on the mailing list are available in SimRel now as well.

**Comment by laeubi** [comment]
I think its always a good idea to have a release before upgrading maven

**Comment by mickaelistria** [comment]
Submitted IP review
```
mistria@localhost:~/git/m2e-core/m2e-maven-runtime/org.eclipse.m2e.maven.runtime$ mvn dependency:list -DexcludeGroupIds=org.eclipse.m2e,p2.eclipse-plugin  -DskipTests -Dmaven.javadoc.skip=true -DappendOutput=true -DoutputFile=maven.deps 
[INFO] Scanning for projects...
[INFO] Computing target platform for MavenProject: org.eclipse.m2e:org.eclipse.m2e.maven.runtime:1.19.0-SNAPSHOT @ /home/mistria/git/m2e-core/m2e-maven-runtime/org.eclipse.m2e.maven.runtime/pom.xml
[INFO] Resolving MavenDependencyRoots = [GroupId = biz.aQute.bnd, ArtifactId = biz.aQute.bndlib, Version = 5.1.2, ArtifactType = jar, IncludeDependencyScope = , GroupId = io.takari.m2e.workspace, ArtifactId = org.eclipse.m2e.workspace.cli, Version = 0.3.1, ArtifactType = jar, IncludeDependencyScope = ], IncludeDependencyScope = null, MissingManifestStrategy = ERROR, IncludeSource = true...
[INFO] Resolving dependencies of MavenProject: org.eclipse.m2e:org.eclipse.m2e.maven.runtime:1.19.0-SNAPSHOT @ /home/mistria/git/m2e-core/m2e-maven-runtime/org.eclipse.m2e.maven.runtime/pom.xml
[WARNING] The following locally built units have been used to resolve project dependencies:
[WARNING]   org.eclipse.m2e.maven.runtime.slf4j.simple/1.18.1.20211008-0657
[INFO] Resolving class path of MavenProject: org.eclipse.m2e:org.eclipse.m2e.maven.runtime:1.19.0-SNAPSHOT @ /home/mistria/git/m2e-core/m2e-maven-runtime/org.eclipse.m2e.maven.runtime/pom.xml
[WARNING] The POM for org.sonarsource.scanner.maven:sonar-maven-plugin:jar:${sonar-maven-plugin.version} is missing, no dependency information available
[WARNING] Failed to retrieve plugin descriptor for org.sonarsource.scanner.maven:sonar-maven-plugin:${sonar-maven-plugin.version}: Plugin org.sonarsource.scanner.maven:sonar-maven-plugin:${sonar-maven-plugin.version} or one of its dependencies could not be resolved: org.sonarsource.scanner.maven:sonar-maven-plugin:jar:${sonar-maven-plugin.version} was not found in https://repo.eclipse.org/content/repositories/tycho-snapshots/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of tycho-snapshots has elapsed or updates are forced
[INFO] 
[INFO] -----------< org.eclipse.m2e:org.eclipse.m2e.maven.runtime >------------
[INFO] Building M2E Embedded Maven Runtime (includes Incubating components) 1.19.0-SNAPSHOT
[INFO] ---------------------------[ eclipse-plugin ]---------------------------
[WARNING] The POM for org.sonarsource.scanner.maven:sonar-maven-plugin:jar:${sonar-maven-plugin.version} is missing, no dependency information available
[WARNING] Failed to retrieve plugin descriptor for org.sonarsource.scanner.maven:sonar-maven-plugin:${sonar-maven-plugin.version}: Plugin org.sonarsource.scanner.maven:sonar-maven-plugin:${sonar-maven-plugin.version} or one of its dependencies could not be resolved: org.sonarsource.scanner.maven:sonar-maven-plugin:jar:${sonar-maven-plugin.version} was not found in https://repo.eclipse.org/content/repositories/tycho-snapshots/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of tycho-snapshots has elapsed or updates are forced
[INFO] 
[INFO] --- maven-dependency-plugin:2.2:list (default-cli) @ org.eclipse.m2e.maven.runtime ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  16.004 s
[INFO] Finished at: 2021-11-23T10:40:02+01:00
[INFO] ------------------------------------------------------------------------
mistria@localhost:~/git/m2e-core/m2e-maven-runtime/org.eclipse.m2e.maven.runtime$ java -jar ~/git/dash-licenses/core/target/org.eclipse.dash.licenses-0.0.1-SNAPSHOT.jar maven.deps -review -project technology.m2e -token $GITLAB_TOKEN
[main] INFO Querying Eclipse Foundation for license data for 42 items.
[main] INFO Found 16 items.
[main] INFO Querying ClearlyDefined for license data for 26 items.
[main] INFO Found 26 items.
License information could not be automatically verified for the following content:

maven/mavencentral/org.apache.maven/maven-compat/3.8.4
maven/mavencentral/org.apache.maven/maven-core/3.8.4
maven/mavencentral/org.apache.maven/maven-embedder/3.8.4
maven/mavencentral/org.apache.maven/maven-plugin-api/3.8.4

This content is either not correctly mapped by the system, or requires review.


Setting up a review for maven/mavencentral/org.apache.maven/maven-compat/3.8.4.
 - Created: https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/1661
Setting up a review for maven/mavencentral/org.apache.maven/maven-plugin-api/3.8.4.
 - Created: https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/1662
Setting up a review for maven/mavencentral/org.apache.maven/maven-core/3.8.4.
 - Created: https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/1663
Setting up a review for maven/mavencentral/org.apache.maven/maven-embedder/3.8.4.
 - Created: https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/1664
```

File: m2e-maven-runtime/org.eclipse.m2e.maven.runtime/pom.xml
Comment (HannesWell) [inline m2e-maven-runtime/org.eclipse.m2e.maven.runtime/pom.xml:29]: The value of the `maven-resolver.version` property has to be set to `1.6.3` as well.

**Review by HannesWell — CHANGES_REQUESTED** [review]

**Comment by HannesWell** [comment]
```
[ERROR] Cannot resolve project dependencies:
[ERROR]   Software being installed: org.eclipse.m2e.archetype.common 1.18.1.qualifier
[ERROR]   Missing requirement: org.eclipse.m2e.archetype.common 1.18.1.qualifier requires 'osgi.bundle; org.eclipse.m2e.maven.runtime [1.18.0,1.19.0)' but it could not be found
```
The required bundle ranges in org.eclipse.m2e.archetype.common have to be adjusted (maven.indexer as well?).

**Comment by mickaelistria** [comment]
Actually, movign to 1.19.0 for an upgrade of a Maven service release is too much, I'll just bump by +0.0.1

**Comment by HannesWell** [comment]
> Actually, movign to 1.19.0 for an upgrade of a Maven service release is too much, I'll just bump by +0.0.1

Make sense.

The `org.eclipse.m2e.feature` and `m2e.sdk` feature need version bumps as well.

**Review by HannesWell — APPROVED** [review]

**Comment by mickaelistria** [comment]
All IP requests are approved, let's merge.

**Pull request merged**
